Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -613,13 +668,22 @@ protected class ClassWriter : IClassWriter {
             switch (cl) {
               case SubsetTypeDecl td:
                 switch (td.WitnessKind) {
+                  case SubsetTypeDecl.WKind.Compiled:
+                    return TypeName_UDT(FullTypeName(udt), udt, wr, udt.tok) + ".default()";
                   case SubsetTypeDecl.WKind.Special:
-                    Contract.Assert(ArrowType.IsPartialArrowTypeName(td.Name) || ArrowType.IsTotalArrowTypeName(td.Name) || td is NonNullTypeDecl);
-                    if (td is NonNullTypeDecl) { return "None"; }
-                    Contract.Assert(udt.TypeArgs.Any());
+                    if (ArrowType.IsPartialArrowTypeName(td.Name)) {
+                      return "None";
+                    }
+                    if (td is NonNullTypeDecl decl) {
+                      if (decl.Class is ArrayClassDecl arr) {
+                        return $"{DafnyArrayClass}.empty({arr.Dims})";
+                      }
+                      return "None";
+                    }
+                    Contract.Assert(udt.TypeArgs.Any() && ArrowType.IsTotalArrowTypeName(td.Name));
                     var rangeDefaultValue = TypeInitializationValue(udt.TypeArgs.Last(), wr, tok, usePlaceboValue,
                       constructTypeParameterDefaultsFromTypeDescriptors);
-                    var arguments = udt.TypeArgs.Comma((_, i) => $"x{i}");
+                    var arguments = udt.TypeArgs.SkipLast(1).Comma((_, i) => $"x{i}");
```

Comment:
Took me a bit to figure out why we needed the `SkipLast`, I assume it's because at this point we know udt is a total arrow and hence we're skipping the result type? Probably worth a quick comment.